### PR TITLE
Fix BlazorServer Identity templates

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Fixes https://github.com/dotnet/aspnetcore/issues/25353

### CODE REVIEW

**Comment by JunTaoLuo** [comment]
FYI @jmprieur @javiercn

**Review by jmprieur — APPROVED** [review]
LGTM
Thanks @JunTaoLuo

**Comment by JunTaoLuo** [comment]
A few notes for reviewers:

1. I verified manually that a blazorserver template with individual auth using local db has now resolved the original issue as described in https://github.com/aspnet/AspNetCore-ManualTests/issues/140. Specifically, now at the email confirmation page:
image
The links Register and Login now work. For example, after I click Login it directs me to this page:
image

2. In verifying the fix I think I uncovered an issue that current breaks blazorserver template for any configuration (Auth, no auth, etc). Specifically, it seems like `@using Microsoft.AspNetCore.Components.Web.Virtualization` was added in https://github.com/dotnet/aspnetcore/blame/release/5.0/src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/_Imports.razor#L7. To my understanding, Virtualization is a purely blazor wasm concern and this using does not belong in a blazor server template. The effect of having such a using statement leads to errors: 
```
C:\gh\tp\BlazorServer\_Imports.razor(7,43): error CS0234: The type or namespace name 'Virtualization' does not exist in the namespace 'Microsoft.AspNetCore.Components.Web' (are you missing an assembly reference?) [C:\gh\tp\BlazorServer\BlazorServer.csproj]
```
If my understanding is correct, I'm very concerned that this issue wasn't caught earlier

3. I'm a little perplexed by the recommendation in the linked issue:
> The fix would be to use a different set of _LoginPartial files, one for OrgAuth and for IndividualAuth similar to the Razor Pages template: https://github.com/dotnet/aspnetcore/blob/master/src/ProjectTemplates/Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Pages/Shared/_LoginPartial.OrgAuth.cshtml.
Specifically, the contents of RazorPagesWeb-CSharp/Pages/Shared/_LoginPartial.OrgAuth.cshtml and the original contents of BlazorServerWeb-CSharp/Identity/Pages/Shared/_LoginPartial.cshtml are quite different, such as the content generated for `IndividualB2CAuth` scenarios. I'm choosing to make the minimum changes to resolve the issue outlined in the original issue https://github.com/aspnet/AspNetCore-ManualTests/issues/140. I'm leaving the option to make further changes to BlazorServerWeb-CSharp/Identity/Pages/Shared/_LoginPartial.cshtml for the next milestone after folks with more context can elaborate on what needs to be updated if any.

**Review by javiercn — APPROVED** [review]
LGTM

**Review by jmprieur — COMMENTED** [review]
LGTM.

Since areas is only used in the IndividualLocalAuth case, maybe just the content of **content/BlazorServerWeb-CSharp/Areas/Identity/Pages/Shared/_LoginPartial.cshtml** with the content you've proposed for **.../content/BlazorServerWeb-CSharp/Areas/Identity/Pages/Shared/_LoginPartial.Identity.cshtml** would work?

**Comment by JunTaoLuo** [comment]
@jmprieur you're absolutely right, It didn't occur to me that we don't need the _LoginPartial.OrgAuth.cshtml at all and took the recommendation in the issue too literally. However, I'd like to addres that for rc2 instead of retriggering the CI checks again, there's a chance they might not complete before the 10AM deadline. Given the current code is correct but just more verbose than necessary, I'd like to get this change merged as-is.

**Comment by Pilchie** [comment]
Approved for RC1 if merged before 10am Pacific on 2020-09-01.

**Comment by jmprieur** [comment]
That sounds good to me, @JunTaoLuo 
I tested the patch on my templates, and this works (that's the way I discovered it was not needed)

**Review by jmprieur — APPROVED** [review]
LGTM
Tested

File: src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/_Imports.razor [hunk 1]
```diff
@@ -4,7 +4,6 @@
 @using Microsoft.AspNetCore.Components.Forms
 @using Microsoft.AspNetCore.Components.Routing
 @using Microsoft.AspNetCore.Components.Web
-@using Microsoft.AspNetCore.Components.Web.Virtualization
```
Comment (captainsafia) [inline src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/_Imports.razor:7]: Why is this removed?

Comment (JunTaoLuo) [reply src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/_Imports.razor:7]: Why was this added in the first place? FYI, I discussed this a little more in https://github.com/dotnet/aspnetcore/pull/25456#issuecomment-684599620.

Comment (pranavkm) [reply src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/_Imports.razor:7]: Virtualization isn't wasm specific, it's part of the core component experience: https://github.com/dotnet/aspnetcore/blob/master/src/Components/Web/src/Virtualization/Virtualize.cs#L12-L18. 

It's weird that you're seeing the error. Perhaps you are using an an old targeting pack?

Comment (captainsafia) [reply src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/_Imports.razor:7]: Yeah, I believe the template is fine here and this is some environment-related issue. I see this PR is already merged but we should consider bringing it back in another PR. Perhaps @MackinnonBuck can resolve this in his virtualization improvements PR?

Comment (JunTaoLuo) [reply src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/_Imports.razor:7]: I'm making a followup PR to this one to rc2 so I'll do it there. Can one of you help me figure out why I was running into the error during verification?
